returnn/torch/engine.py

```diff
@@ -1475,17 +1498,23 @@ def _print_process(
             info += ["%.3f sec/step" % step_duration]
         if start_elapsed is not None:
             info += ["elapsed %s" % hms(start_elapsed)]
-        if num_seqs is not None:
-            assert seq_idx is not None and start_elapsed is not None  # unexpected combination...
+        if complete_frac is not None:
+            complete = complete_frac
+        elif num_seqs is not None:
```

Comment:
Do we really need to have this logic here? How can it be that `complete_frac` is None but `num_seqs` is known? I think, if `num_seqs` is known, `complete_frac` should never be None. This should be handled properly anywhere in the pipeline, or at least e.g. in `ReturnnDatasetIterDataPipe`.